### PR TITLE
✨use legacy successors, support skips and skipRange

### PR DESCRIPTION
**Pull request by joelanford**
# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/catalogmetadata/filter/bundle_predicates.go [hunk 1]
```diff
@@ -58,11 +59,30 @@ func WithBundleImage(bundleImage string) Predicate[catalogmetadata.Bundle] {
...
+			return true
+		}
+		for _, skip := range candidateBundleEntry.Skips {
+			if skip == installedBundle.Name {
+				return true
+			}
+		}
+		if candidateBundleEntry.SkipRange != "" {
+			installedBundleVersion, _ := installedBundle.Version()
+			skipRange, _ := bsemver.ParseRange(candidateBundleEntry.SkipRange)
```
Comment (grokspawn) [inline internal/catalogmetadata/filter/bundle_predicates.go:74]: The difference between blang semver and masterminds semver import aliases is so little that overlooking the subtlety is really easy to do.  
Your code is fine, though I can't help but wonder though when someone will 'cross the streams' and we'll miss it.

Comment (joelanford) [reply internal/catalogmetadata/filter/bundle_predicates.go:74]: Yeah agreed. Perhaps we should think about adding some wrappers that obviate the subtlety?

**Review by everettraven — APPROVED** [review]
LGTM!

**Comment by grokspawn** [comment]
Solves #54

**Pull request merged**